dds/DCPS/QOS_XML_Handler/QOS_PubSub_T.cpp

```diff
@@ -50,84 +50,84 @@ QOS_PubSub_T<XML_QOS_TYPE, DDS_QOS_TYPE>::read_qos (DDS_QOS_TYPE& dds_qos, const
-                  ACE_TEXT ("Set presentation access scope to <%d>\n"),
+                ACE_DEBUG((LM_TRACE,
+                  ACE_TEXT("QOS_PubSub_T<XML_QOS_TYPE, DDS_QOS_TYPE>::read_qos - ")
+                  ACE_TEXT("Set presentation access scope to <%d>\n"),
                   dds_qos.presentation.access_scope));
               }
           }
-        if (xml_qos->presentation ().coherent_access_p ())
+        if (xml_qos->presentation().coherent_access_p())
           {
             dds_qos.presentation.coherent_access =
-              xml_qos->presentation ().coherent_access ();
+              xml_qos->presentation().coherent_access();
 
             if (OpenDDS::DCPS::DCPS_debug_level > 9)
               {
-                ACE_DEBUG ((LM_TRACE,
-                  ACE_TEXT ("QOS_PubSub_T<XML_QOS_TYPE, DDS_QOS_TYPE>::read_qos - ")
-                  ACE_TEXT ("Set presentation coherent_access to <%d>\n"),
+                ACE_DEBUG((LM_TRACE,
+                  ACE_TEXT("QOS_PubSub_T<XML_QOS_TYPE, DDS_QOS_TYPE>::read_qos - ")
+                  ACE_TEXT("Set presentation coherent_access to <%d>\n"),
                   dds_qos.presentation.coherent_access));
               }
           }
-        if (xml_qos->presentation ().ordered_access_p ())
+        if (xml_qos->presentation().ordered_access_p())
           {
             dds_qos.presentation.ordered_access =
-              xml_qos->presentation ().ordered_access ();
+              xml_qos->presentation().ordered_access();
 
             if (OpenDDS::DCPS::DCPS_debug_level > 9)
               {
-                ACE_DEBUG ((LM_TRACE,
-                  ACE_TEXT ("QOS_PubSub_T<XML_QOS_TYPE, DDS_QOS_TYPE>::read_qos - ")
-                  ACE_TEXT ("Set presentation ordered_access to <%d>\n"),
+                ACE_DEBUG((LM_TRACE,
+                  ACE_TEXT("QOS_PubSub_T<XML_QOS_TYPE, DDS_QOS_TYPE>::read_qos - ")
+                  ACE_TEXT("Set presentation ordered_access to <%d>\n"),
                   dds_qos.presentation.ordered_access));
               }
           }
       }
 
-  if (xml_qos->partition_p ())
+  if (xml_qos->partition_p())
     {
-      if (xml_qos->partition ().name_p ())
+      if (xml_qos->partition().name_p())
         {
           dds_qos.partition.name.length(static_cast<CORBA::ULong>(xml_qos->partition().name().count_element()));
           CORBA::ULong pos = 0;
-          for (::dds::stringSeq::element_const_iterator it = xml_qos->partition ().name ().begin_element ();
-               it != xml_qos->partition ().name ().end_element ();
+          for (::dds::stringSeq::element_const_iterator it = xml_qos->partition().name().begin_element();
+               it != xml_qos->partition().name().end_element();
                ++it, ++pos)
             {
-              dds_qos.partition.name[pos] = ::CORBA::string_dup (it->get()->c_str());
+              dds_qos.partition.name[pos] = ACE_NS_WString(it->get()->c_str()).char_rep();
```

Comment:
Why is this `ACE_NS_WString`?  It's not clear to me why that would be used over one of the other converters.

Reply:
I think this was early in my efforts, didn't realize ACE_TEXT_ALWAYS_* macros existed.